### ISSUE TITLE
Errors when opening database after Ubuntu update

### ISSUE THREAD

**Issue by markmaker** [pwsafe/pwsafe#873]
After updating Ubuntu to 22.04, where I assume `pwsafe` (installed through `apt`) got upgraded as well, I get a few of these errors when opening:

![Screenshot_20220815_082701](https://user-images.githubusercontent.com/9963310/184591076-8c8c1aa9-72b4-481b-bbc1-1b4fea398de4.png)

> Read Error: Trouble with a non-textual (e.g. time) field in record 'xxx' - please check data carefully.

After OKing these, I get some corrupted entries. They also lost their group assignment.

I restored a backup to exclude corruption of the database itself. Same result. 

Unfortunately, I don't know the version I had before. 

Thanks for all help, 
Mark

**Comment by markmaker** [comment]
OK, I seem to have found the reason myself. Using KDE System Settings, I had set the time format to the regional locale, that seems to break the data base format. Setting it to "C (default)" fixed this. 

I still consider this to be a bug. Times should be persisted in a neutral machine-readable format, right? 

Thanks for looking into it.

**Comment by rafaelx** [comment]
Could you please tell me what your locale is or what you have switched to in the settings?

**Comment by markmaker** [comment]
I switched to something like `en_ch-utf8` i.e. "English Language Swiss Format with UTF-8" (I'm not currently at that computer and can't check the exact name, sorry).

**Comment by rafaelx** [comment]
So far I haven't been able to reproduce the problem on 20.04.01 LTS with the latest version of pwsafe. Maybe I'm missing something crucial.

If I'm not mistaken, the official repositories for [jammy](https://packages.ubuntu.com/jammy/passwordsafe) provide version 1.12. Would you have the possibility to update to the latest version 1.15, which can be found at [GitHub](https://github.com/pwsafe/pwsafe/releases)?

If the problem is also reproducible with version 1.15, could you please provide the output of the _locale_ command?

**Comment by markmaker** [comment]
Hi @rafaelx,

thanks for following up on this. 

Now that you wrote back, I have investigated some more. I tried pwsafe from the shell and got:

```
$ pwsafe
Couldn't load locale, falling back to default
Current locale doesn't support UTF-8
Couldn't detect locale. Trying to skip empty env. variables.

```
Then I found this:

https://askubuntu.com/questions/162391/how-do-i-fix-my-locale-issue

I think this is specific to an upgraded Ubuntu system, some things are missing in the Ubuntu upgrade process (see some comments to that effect there). 

I followed the advice in both the [first](https://askubuntu.com/a/227513/1531206) and [second](https://askubuntu.com/a/229512/1531206) answer. Logged out and in again. And then it worked, even if I set the time format back to Swiss (don't like the AM/PM, actually).

I think `pwsafe` is not to blame, at least not for the _trigger_ of the error. Hrm... as a software developer myself, I still think a database should always store data in a locale-neutral machine-readable format, hehe. 🤓

_Mark
